### PR TITLE
💄(jitsi-box:student) Display the secondary display as a floating draggable box

### PR DESCRIPTION
**Pull request by faydef**
choose a display(image or video) as your main display and the other as a seconday, therefore it is put in a floating box that you can drag to keep seeing without disturbing the main display, you can switch between diplay modes with a click of a button

### CODE REVIEW

File: box-ui/src/components/ts/FloatingBox.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+const FloatingBox = ({ children }: { children: any }) => {
+    return (
+        <div>
+            <Draggable>
+                <div className='pipWindow'>{children}</div>
+            </Draggable>
+        </div>
+    );
+};
```
Comment (AladdinVR) [inline box-ui/src/components/ts/FloatingBox.tsx]:
```suggestion
const FloatingBox = ({ props }: FloatingBoxProps) => {
    return (
        <div>
            <Draggable>
                <div className='pipWindow'>{props.children}</div>
            </Draggable>
        </div>
    );
};
```

You should create a type for your props as it is done for other components in utils/props

File: box-ui/src/components/css/FloatingBox.css [hunk 2]
```diff
@@ -0,0 +1,26 @@
+.pipWindow {
+    background-color: #242d36;
+    z-index: 100;
+    position: fixed;
+    font-size: 14px;
+    font-weight: 200;
+    line-height: 1.5;
+    color: rgb(248, 248, 249);
```
Comment (djexvr) [inline box-ui/src/components/css/FloatingBox.css]: You don't need to modify the text css

File: box-ui/src/components/css/FloatingBox.css [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    box-sizing: content-box;
+    box-shadow: 1px 3px 3px 0 rgb(0 0 0 / 20%), 1px 3px 15px 2px rgb(0 0 0 / 20%);
+    border-radius: 0.28571429rem;
```
Comment (djexvr) [inline box-ui/src/components/css/FloatingBox.css]: This is very specific

File: box-ui/src/components/ts/FloatingBox.tsx (hunk 1 above)
Comment (djexvr) [reply box-ui/src/components/ts/FloatingBox.tsx]: If you only need children you can only add FunctionComponent to make typescript happy and then get children from props

File: box-ui/src/components/css/FloatingBox.css [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+.pipWindow:hover {
+    cursor: grab;
+}
+
+.pipWindow:active {
+    cursor: grabbing;
+    outline: -webkit-focus-ring-color auto 1px;
+    outline-color: -webkit-focus-ring-color;
+    outline-style: auto;
+    outline-width: 1px;
```
Comment (djexvr) [inline box-ui/src/components/css/FloatingBox.css]: You're repeating yourself the line 22 should suffice.

`-webkit-focus-ring-color` is only avalaible on safari you should add for firefox and chrome
```suggestion
    outline: -webkit-focus-ring-color auto 1px;
    outline: Highlight auto 1px;
```

Comment (djexvr) [inline box-ui/src/components/css/FloatingBox.css]: You're repeating yourself the line 22 should suffice.

`-webkit-focus-ring-color` is only avalaible on safari you should add for firefox and chrome
```suggestion
    outline: -webkit-focus-ring-color auto 1px;
    outline: Highlight auto 1px;
```

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 5]
```diff
@@ -191,22 +200,30 @@ const StudentMeeting: FunctionComponent = () => {
...
     }
 
+    function ChangeView() {
+        setMiniImg((miniImg) => !miniImg);
+    }
+
+    function ChangeMinimize() {
+        setMinimize((minimize) => !minimize);
+    }
+
```
Comment (djexvr) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx:211]: Do you really need a function for that can't you use it directly ?

File: box-ui/src/components/ts/FloatingBox.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import Draggable from 'react-draggable';
+import '../css/FloatingBox.css';
+
+const FloatingBox = ({ children }: { children: any }) => {
+    return (
+        <div>
```
Comment (djexvr) [inline box-ui/src/components/ts/FloatingBox.tsx]: Do you need this div. It's better to not add unnecessary div.

File: box-ui/src/utils/Props/index.d.ts [hunk 7]
```diff
@@ -67,3 +67,7 @@ export type FocusModeProps = {
     focusItems: { element: string; textElement: string }[];
     setDisplayFocus: React.Dispatch<React.SetStateAction<boolean>>;
 };
+
+export type FloatProps = {
```
Comment (djexvr) [inline box-ui/src/utils/Props/index.d.ts]: You don't need to do this if you create it with FunctionComponent

**Comment by djexvr** [comment]
I really think you should create a component for the Image Viewer and use it in StudentMeeting rather than recoding it in every floating or non-floating window it is really the opposite of the react philosophy and will make modifying the interface way more complicated and error-prone.

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 8]
```diff
@@ -279,99 +280,35 @@ const StudentMeeting: FunctionComponent = () => {
...
-                                maxWidth: '45vw',
-                            }}
-                        ></ClickableSVG>
+                        <ImageViewer
+                            img1={[img, '30vh', ((widthImgDouble / 45) * 30).toString() + 'vh']}
+                            img2={[imgOriginal, '30vh', ((ratioImgOriginal / 45) * 30).toString() + 'vh']}
+                            onclick={addCircle}
+                            addOn={circles}
+                            selectWindow={selectCoord}
+                        ></ImageViewer>
```
Comment (djexvr) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx]: You can use an orphan tag `<ImageViewer ...... />`

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 9]
```diff
@@ -420,49 +357,13 @@ const StudentMeeting: FunctionComponent = () => {
...
-                                backgroundSize: 'contain',
-                                maxWidth: '40vw',
-                            }}
-                        ></ClickableSVG>
-                    </div>
+                    <ImageViewer
+                        img1={[img, '45vh', widthImgDouble.toString() + 'vh']}
+                        img2={[img, '45vh', ratioImgOriginal.toString() + 'vh']}
+                        selectWindow={selectCoord}
+                    ></ImageViewer>
```
Comment (djexvr) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx]: Same as before use `<ImageViewer ..../>`

File: box-ui/src/components/ts/FloatingBox.tsx [hunk 10]
```diff
@@ -5,11 +5,9 @@ import { FloatProps } from '../../utils/Props';
 
 const FloatingBox = ({ children }: FloatProps) => {
```
Comment (djexvr) [inline box-ui/src/components/ts/FloatingBox.tsx]: use `const FloatingBox: FunctionComponent = (props) =>` to be consistent with the other components.

**Comment by djexvr** [comment]
Nice work well done 👍

**Pull request merged**